numpy/core/_add_newdocs.py

```diff
@@ -7969,62 +7969,199 @@ def luf(lamdaexpr, *args, **kwargs):
 #
 ##############################################################################
 
-add_newdoc('numpy.core.numerictypes', 'bool_',
-    """NumPy's Boolean type.  Character code: ``?``.  Alias: bool8""")
+def get_type(place, obj):
+    try:
+        return getattr(__import__(place, globals(), {}, [obj]), obj)
+    except Exception:
+        return None
+
+def type_aliases(place, aliases):
+    return [(alias_type, doc) for (alias, doc) in aliases
+                for alias_type in (get_type(place, alias),)
+                if alias_type is not None]
+
+integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('int8', '8-bit singed integer (-128 to 127)'),
+    ('int16', '16-bit singed integer (-32768 to 32767)'),
+    ('int32', '32-bit singed integer (-2147483648 to 2147483647)'),
+    ('int64', '64-bit singed integer (-9223372036854775808 to 9223372036854775807)'),
+    ('intp', 'Signed integer large enough to fit pointer, compatible with C ``ssize_t``'),
+    ])
+
+unsigned_integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('uint8', '8-bit unsinged integer (0 to 255)'),
+    ('uint16', '16-bit unsinged integer (0 to 65535)'),
+    ('uint32', '32-bit unsinged integer (0 to 4294967295)'),
+    ('uint64', '64-bit unsinged integer (0 to 18446744073709551615)'),
+    ('uintp', 'Unsigned integer large enough to fit pointer, compatible with C ``size_t``'),
```

Comment:
Strictly it's `uintptr_t`

Reply:
Changed `size_t` and `ssize_t` to `intptr_t` and `uintptr_t`, but the reference docs are again slightly wrong, here, then: https://docs.scipy.org/doc/numpy/user/basics.types.html

Should these be fixed as well?